contrib/IECoreAlembic/src/IECoreAlembic/AlembicScene.cpp

```diff
@@ -367,145 +367,72 @@ class AlembicScene::AlembicReader : public AlembicIO
-
-			if ( !propertyReader )
+			if( !propertyReader )
 			{
-				return 0;
+				return nullptr;
 			}
 
-			ScalarPropertyReaderPtr scalarPropertyReader = propertyReader->getScalarProperty( name.string() );
-
-			if( !scalarPropertyReader )
-			{
-				return 0;
-			}
+			return propertyReader->getScalarProperty( name.string() );
+		}
 
-			return scalarPropertyReader->getNumSamples();
+		size_t numAttributeSamples( const Name &name ) const
+		{
+			ScalarPropertyReaderPtr reader = scalarPropertyReader( name );
+			return reader ? reader->getNumSamples() : 0;
 		}
 
 		double attributeSampleTime( const Name &name, size_t sampleIndex ) const
 		{
-			if( !m_xform )
+			ScalarPropertyReaderPtr reader = scalarPropertyReader( name );
+			if( !reader )
 			{
 				return 0.0;
 			}
-
-			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
-
-			if ( !userProperties.valid() )
-			{
-				return 0;
-			}
-
-			Abc::CompoundPropertyReaderPtr propertyReader = GetCompoundPropertyReaderPtr( userProperties );
-
-			if ( !propertyReader )
-			{
-				return 0;
-			}
-
-			ScalarPropertyReaderPtr scalarPropertyReader = propertyReader->getScalarProperty( name.string() );
-
-			if ( !scalarPropertyReader )
-			{
-				return 0;
-			}
-			auto timeSampling = scalarPropertyReader->getTimeSampling();
+			auto timeSampling = reader->getTimeSampling();
 			return timeSampling->getSampleTime( sampleIndex );
 		}
 
 		double attributeSampleInterval( const Name &name, double time, size_t &floorIndex, size_t &ceilIndex ) const
 		{
-			if( !m_xform )
-			{
-				return 0.0;
-			}
-
-			ICompoundProperty userProperties = m_xform.getSchema().getUserProperties();
-
-			if ( !userProperties.valid() )
-			{
-				return 0;
-			}
-
-			Abc::CompoundPropertyReaderPtr propertyReader = GetCompoundPropertyReaderPtr( userProperties );
-
-			if ( !propertyReader )
+			ScalarPropertyReaderPtr reader = scalarPropertyReader( name );
+			if( !reader )
 			{
 				return 0.0;
 			}
-
-			ScalarPropertyReaderPtr scalarPropertyReader = propertyReader->getScalarProperty( name.string() );
-
-			if ( !scalarPropertyReader )
-			{
-				return 0.0;
-			}
-
-			return sampleInterval( scalarPropertyReader->getTimeSampling().get(), scalarPropertyReader->getNumSamples(), time, floorIndex, ceilIndex );
+			return sampleInterval( reader->getTimeSampling().get(), reader->getNumSamples(), time, floorIndex, ceilIndex );
 		}
 
 		IECore::ConstObjectPtr readAttributeAtSample( const Name &name, size_t sampleIndex ) const
 		{
-			if( !m_xform )
-			{
-				return nullptr;
-			}
-
-			const IXformSchema &schema = m_xform.getSchema();
-			auto userProperties = schema.getUserProperties();
-
-			const PropertyHeader *propertyHeader = userProperties.getPropertyHeader( name.string() );
-			if( !propertyHeader )
-			{
-				return nullptr;
-			}
-
-			if( propertyHeader->getPropertyType() != kScalarProperty )
```

Comment:
It appears we no longer have the explicit check against kScalarProperty.  I guess this now handled by getScalarProperty returning null?

Reply:
Yes, this is confirmed by the Alembic docs for `CompoundPropertyReader::getScalarProperty()` :

> It will return an empty pointer if the property is not scalar or is not found.